### PR TITLE
Add state persistence hooks for table preferences

### PR DESCRIPTION
**Pull request by gilgardosh**
## Summary

Introduces two new React hooks for managing component state with different persistence strategies, and applies them to the businesses table to preserve user preferences (sorting, column visibility) across page refreshes while keeping row selection transient.

## Key Changes

- **`usePersistentState` hook** (`packages/client/src/hooks/use-persistent-state.ts`): A `useState` wrapper that mirrors state to `localStorage`. Includes a `revive` callback to validate and transform stored values (handling stale/corrupt entries from older app versions). Gracefully falls back to the fallback value if storage is unavailable or parsing fails.

- **`useEphemeralState` hook** (`packages/client/src/hooks/use-ephemeral-state.ts`): A `useState` wrapper that keeps state in a module-level store, surviving unmount/remount (navigation) but not page refreshes. Intended for transient selections that should not persist across reloads.

- **Table state storage utilities** (`packages/client/src/components/businesses/table-state-storage.ts`): Defensive revivers for sorting and column visibility state that validate stored shapes and merge with current defaults, allowing graceful handling of schema changes and missing columns.

- **Comprehensive test coverage**: Unit tests for both hooks and the storage utilities, covering happy paths, error cases, and edge cases (corrupt JSON, reviver failures, storage quota exceeded, etc.).

- **Businesses table integration**: Replaced plain `useState` calls with the new hooks:
  - Row selection uses `useEphemeralState` (transient across navigation, cleared on refresh)
  - Column visibility and sorting use `usePersistentState` (preserved across refreshes)
  - Fixed row selection to track the unfiltered row set so selections persist when filters change

## Notable Implementation Details

- Storage access is fully guarded: failures in `localStorage` or JSON parsing fall back gracefully without crashing the screen.
- The `revive` callback enables safe schema evolution—stored values from older app versions are validated and merged with current defaults rather than blindly restored.
- Row selection now correctly uses the full unfiltered row set rather than just visible rows, so selections survive filter changes and are re-checked when the filter is removed.

https://claude.ai/code/session_01Pwx8BHZDnssEpQJDmDyvbv

### CODE REVIEW

File: packages/client/src/hooks/use-persistent-state.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  });
+
+  useEffect(() => {
+    try {
+      localStorage.setItem(storageKey, JSON.stringify(state));
+    } catch (error) {
+      console.warn(`Failed to persist "${storageKey}" to localStorage:`, error);
+    }
+  }, [storageKey, state]);
```
[the remark this answers is not in the document]
Comment (gilgardosh) [reply packages/client/src/hooks/use-persistent-state.ts]: Fixed in fd99740. Reading now lives in a `readStored` helper, used by both the `useState` initializer and a new key-change path: the hook tracks the key it hydrated from and, when `storageKey` changes, re-reads that key and returns the fresh value in the same render pass (updating state during render, as `useStableValue` already does in this codebase). Added two tests — one asserting the new key's stored value wins and the old key's value isn't written under it, one asserting an unstored key falls back.

The current call sites all pass constant keys, so this wasn't reachable in practice, but it's a real footgun for a derived (per-user/per-screen) key — worth closing.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/client/src/components/businesses/__tests__/table-state-storage.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+describe('reviveColumnVisibility', () => {
+  it('merges the stored value over the defaults', () => {
+    expect(reviveColumnVisibility({ name: false }, VISIBILITY_FALLBACK)).toEqual({
+      name: false,
+      city: false,
+    });
+  });
+
+  it('keeps defaults for columns added since the value was stored', () => {
```
[the remark this answers is not in the document]
Comment (gilgardosh) [reply packages/client/src/components/businesses/__tests__/table-state-storage.test.ts]: Fixed in fd99740. The one case was checking two behaviours at once, so I split it: "keeps the default for a column added since the value was stored" now uses a stored value with only `name` (so `city` keeps its hidden default), and "carries over a stored entry for a column that no longer exists" covers the `gone` key, with a comment on why keeping it is intentional — the table ignores visibility entries with no matching column, and retaining the entry means a column that comes back keeps the visibility the user last chose for it.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/client/src/hooks/__tests__/use-ephemeral-state.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    current: () => state,
+    set: value => act(() => setState(value)),
+    unmount: () => act(() => localRoot.unmount()),
+  };
+}
+
+describe('useEphemeralState', () => {
+  it('starts from the fallback', () => {
+    expect(mount({ a: true }).current()).toEqual({ a: true });
+  });
```
[the remark this answers is not in the document]
Comment (gilgardosh) [reply packages/client/src/hooks/__tests__/use-ephemeral-state.test.ts:60]: Fixed in fd99740. `mount` now pushes each root into a `mountedRoots` array and `afterEach` unmounts everything still live, so the roots created by the assertions on lines 50/57/75 (which deliberately never call `unmount`) are torn down. Calling `unmount` explicitly still removes the root from the list, so it isn't unmounted twice.

---
_Generated by [Claude Code](https://claude.ai/code)_
